bap-vibes/src/substituter.ml

```diff
@@ -73,39 +72,119 @@ let size_of_typ (typ : typ) (name : string) =
       (Subst_err
          (Format.sprintf "Unexpected type for variable: %s" name))
 
+(* Initialize variables with their `at-entry` values. *)
+let initialize
+    (blks : blk term list)
+    ~(typeof : string -> typ option)
+    ~(entry_tid : tid)
+    ~(hvars : Hvar.t list)
+    ~(tgt : Theory.target)
+    ~(spilled : String.Set.t) : blk term list KB.t =
+  KB.List.map blks ~f:(fun blk ->
+      let tid = Term.tid blk in
+      if Tid.(tid = entry_tid) then
+        let+ defs = KB.List.filter_map hvars ~f:(fun hvar ->
+            let name = Hvar.name hvar in
+            if Set.mem spilled name then KB.return None
+            else match Hvar.value hvar with
+              | Hvar.Constant _ -> KB.return None
+              | Hvar.Memory _ -> KB.return None
+              | Hvar.Registers {at_entry = None; _} -> KB.return None
+              | Hvar.Registers {at_entry = Some reg; _} ->
+                match typeof name with
+                | None -> KB.return None
```

Comment:
Can this `None` case ever happen for legitimate reasons?  I'm wondering if we want an error instead of just ignoring this hvar.

Reply:
It could happen if we included some variable in the `patch-vars` but then never used it in the actual patch code, which is why I didn't think to error immediately.

Reply:
Sure, should that be its own PR? I can open a ticket.

Reply:
Whatever you prefer.  It's fine to do it here, if it's easy.